consensus/scheme/rolldpos/rolldpos.go

```diff
@@ -271,6 +272,12 @@ func (b *Builder) SetChainManager(chain ChainManager) *Builder {
 	return b
 }
 
+// SetEvmNetworkID sets the evm network ID context
+func (b *Builder) SetEvmNetworkID(evmNetworkID uint32) *Builder {
```

Comment:
```
func (b *Builder) SetBlockDeserializer(deser *block.Deserializer) *Builder {
    b.blockDeserializer  = deser
    return b
}
```

Reply:
can still use `SetEvmNetworkID`, and create the deserializer internally
```
b.deser = block.NewDeserializer(evmNetworkID)
```